### PR TITLE
Ci script improvements

### PR DESCRIPTION
**Pull request by psychon**
Some stuff, the main thing being that I can run `tests/run.sh` now, too, even though I don't have an up-to-date version of awesome installed.

### CODE REVIEW

File: tests/run.sh [hunk 1]
```diff
@@ -62,47 +60,68 @@ XDG_CONFIG_HOME="./"
...
+    max_wait=60
+    while true; do
+        set +e
+        eval reply="\$($2)"
+        ret=$?
+        set -e
+        if [ $ret = 0 ]; then
+            break
+        fi
+        max_wait=$(expr $max_wait - 1)
```
Comment (blueyed) [inline tests/run.sh]: That needs an `|| true`: `expr 1 - 1` will exit with code 1.
See 938987a1.

File: tests/run.sh [hunk 2]
```diff
@@ -62,47 +60,68 @@ XDG_CONFIG_HOME="./"
...
+        if [ "$max_wait" -lt 0 ]; then
```
Comment (blueyed) [inline tests/run.sh]: ..or use `-le 1` here?!

File: tests/run.sh [hunk 3]
```diff
@@ -62,47 +60,68 @@ XDG_CONFIG_HOME="./"
...
-done
+wait_until_success "setup xrdb" "echo 'Xft.dpi: 96' | DISPLAY='$D' xrdb 2>&1"
 
+# Use a separate D-Bus session; sets $DBUS_SESSION_BUS_PID.
+eval $(DISPLAY="$D" dbus-launch --sh-syntax --exit-with-session)
 
-AWESOME_CLIENT="$root_dir/utils/awesome-client"
+# Not in Travis?
+if [ "$CI" != true ]; then
+    # Prepare a config files pointing to a working theme
```
Comment (blueyed) [inline tests/run.sh]: s/files/file/

**Comment by psychon** [comment]
I rebased this onto your "add lgi 0.7.0"-branch. My plan is to wait until that other branch is merged and then rebase this onto master again and only merge this then.

Note that I removed your `kill -0` trick again. The timeout handles this situation as well and it does not work for my helper function when it is used via `xrdb` to wait for the X11 server to start up.

**Comment by psychon** [comment]
Rebased, can this be merged?

**Pull request merged**